File: frontend/src/types/seating.ts
export interface SeatingAssignmentDetail {
  id: number;
  guest_id: number;
  guest_nama: string;
  jumlah_kursi: number;
}

export interface SeatingTable {
  id: number;
  session_id: string;
  nama_meja: string;
  kapasitas: number;
  kursi_terpakai: number;
  assignments: SeatingAssignmentDetail[];
}

export interface SeatingTableCreateRequest {
  session_id: string;
  nama_meja: string;
  kapasitas?: number;
}

export interface SeatingTableUpdateRequest {
  nama_meja?: string;
  kapasitas?: number;
}

export interface SeatingAssignRequest {
  session_id: string;
  table_id: number;
  guest_id: number;
  jumlah_kursi?: number;
}

export interface UnassignedGuest {
  guest_id: number;
  nama_tamu: string;
  sisa_kursi: number;
}

File: backend/app/schemas/seating.py
from pydantic import BaseModel
from typing import Optional, List


class SeatingAssignmentDetailSchema(BaseModel):
    id: int
    guest_id: int
    guest_nama: str
    jumlah_kursi: int

    class Config:
        from_attributes = True


class SeatingTableSchema(BaseModel):
    id: int
    session_id: str
    nama_meja: str
    kapasitas: int
    kursi_terpakai: int
    assignments: List[SeatingAssignmentDetailSchema] = []

    class Config:
        from_attributes = True


class SeatingTableCreateRequest(BaseModel):
    session_id: str
    nama_meja: str
    kapasitas: int = 8


class SeatingTableUpdateRequest(BaseModel):
    nama_meja: Optional[str] = None
    kapasitas: Optional[int] = None


class SeatingAssignRequest(BaseModel):
    session_id: str
    table_id: int
    guest_id: int
    jumlah_kursi: Optional[int] = None 


class UnassignedGuestSchema(BaseModel):
    guest_id: int
    nama_tamu: str
    sisa_kursi: int